File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.ts
function minElement(nums: number[]): number {
    let minSum = Infinity;

    for (let num of nums) {
        let digitSum = 0;
        let temp = num;
        
        // 각 자릿수의 합 계산
        while (temp > 0) {
            digitSum += temp % 10;
            temp = Math.floor(temp / 10);
        }

        // 최솟값 갱신
        minSum = Math.min(minSum, digitSum);
    }

    return minSum;
}

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        int minSum = Integer.MAX_VALUE;
        
        for (int num : nums) {
            int digitSum = 0;
            int temp = num;
            
            // Calculate the sum of digits
            while (temp > 0) {
                digitSum += temp % 10;
                temp /= 10;
            }
            
            // Update the global minimum
            minSum = Math.min(minSum, digitSum);
        }
        
        return minSum;
    }
}
